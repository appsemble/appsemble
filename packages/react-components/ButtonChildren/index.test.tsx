import { render, screen } from '@testing-library/react';
import { expect, it } from 'vitest';

import { ButtonChildren } from './index.js';

it('should render every child of the component', () => {
  render(
    <div role="tab">
      <ButtonChildren>
        <div>test div</div>
        <button type="button">test button</button>
        <input placeholder="test-input" />
      </ButtonChildren>
    </div>,
  );
  const buttonChildren = screen.getByRole('tab');
  expect(buttonChildren).toMatchSnapshot();
});

it('should render an icon', () => {
  render(
    <div role="tab">
      <ButtonChildren icon="home" iconPosition="left">
        <div>test div</div>
      </ButtonChildren>
    </div>,
  );
  const icon = screen.getByRole('tab');
  expect(icon).toMatchSnapshot();
});
