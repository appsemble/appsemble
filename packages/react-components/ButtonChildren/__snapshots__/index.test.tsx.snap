// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render an icon 1`] = `
<div
  role="tab"
>
  <span
    class="icon"
  >
    <i
      class="fas fa-home"
    />
  </span>
  <span>
    <div>
      test div
    </div>
  </span>
</div>
`;

exports[`should render every child of the component 1`] = `
<div
  role="tab"
>
  <div>
    test div
  </div>
  <button
    type="button"
  >
    test button
  </button>
  <input
    placeholder="test-input"
  />
</div>
`;
